### ISSUE TITLE
World seed value seemingly gets ignored if you use "Start Playing" from Advanced Game Setup

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#3883]
### What you were trying to do

Start the game with the exact same seed to get the same world - got lazy and tried to start playing straight off the Advanced Game Setup (AGS) screen after having entered the seed value

### What actually happened

Got a different world than expected. Found that if I used "Continue to Universe Setup" and manually added my Perlin world then it worked fine. If you use the button on the AGS page it does seem to pick the default world for the gameplay template, but it skips the seed value somehow.

### How to reproduce

* Create a new game, enter a seed value on the AGS page or just copy the default to keep handy
* Continue through remaining steps adding a Perlin world manually and entering the world
* Observe the splendor of your chosen world
* Restart the game client (possible session bleed if you just go back to the main menu, avoid to be sure)
* Enter the same exact seed value on the AGS page, but this time then press "Start Playing"
* Observe a completely different world

I did this with GooeysQuests, but expect it'll affect any set of modules

### Troubleshooting tip

Start by examining what actions happen based on the buttons clicked (search for the button captions if needed to find the associated files). How does the flow vary if you start the game from the AGS page rather than go all the way through the create game flow?

**Comment by AlexJBlair** [comment]
@Cervator Hello, I am fairly new to open source but I found this defect and thought I would give it a try. I was wondering though what is a Perlin world? Is this an extra module?

**Comment by AlexJBlair** [comment]
Hello @Cervator I would like to take a stab at this defect if possible.

**Comment by Cervator** [comment]
Hi @AlexJBlair and welcome! Please do, no need to ask first, although knowing somebody is taking a look is always nice 👍

**Comment by AlexJBlair** [comment]
@Cervator sorry if this is a dumb question but I am trying to understand the reproduction steps a bit. So I was able to reproduce the issue but I was wondering what exactly a "perlin" world is and if this is key to reproducing the issue. 

Thanks

**Comment by Cervator** [comment]
@AlexJBlair please do ask! Although it may be easier to get answers faster on our Discord :-)

I only just so happened to use Perlin. I don't know if it is a requirement, but I somewhat doubt it. I expect it'll be any world that uses the seed (a few might not - for instance GooeyDefence always uses the same world)

You could try other world types to see 👍

**Comment by AlexJBlair** [comment]
I have not been able to fix this issue. Since I am so new to open source and industry, I think I am going to wait and see if there is anyone else who may have a solution and I will follow along to see if I can learn something new and where to look.

**Comment by pollend** [comment]
@AlexJBlair you should jump on the discord and I think people there can offer some guidance. at a minimum you should have a workspace setup with terasology and I think there are a couple ways you can trace the problem. my favorite is to find a working commit and use git bisect to track down the commit that broke the feature. In this case you can pinpoint why the seed isn't being set from the UI. some oversight between the UI and how the seed is being configured. 

I'm sure there are some other options you can look at with getting started. there is some pretty good tutorial documentation in the wiki with regards to getting started and setting things up. create a new module, add a block etc ... 

http://discord.gg/Terasology

**Comment by Cervator** [comment]
Hey @AlexJBlair no problem at all. This issue is a bit detective work'ish and for once I'm not sure if the git bisect will work as I'm not sure this actually ever worked 😁 

Michael is correct about Discord, way easier to help there and guide to other maybe more approachable issues 👍

**Comment by stefaniamak** [comment]
### My theory

I believe the issue here is that the original seed changes value when advancing through the remaining steps (as adding a Perlin world manually).

### What drove me to this conclusion

I worked on issue #3915 and have the seed appear at the Details page. The results I got when following the How to reproduce steps of this (#3883) issue, where: 
* Custom seed value at Advanced Settings Page:
<img width="1015" alt="Screenshot_2" src="https://user-images.githubusercontent.com/48293545/80873276-04c74880-8cc0-11ea-91ca-09ef9fb04c3d.png">

* Seed value at the Game details page after creating the world:
<img width="187" alt="Screenshot_1" src="https://user-images.githubusercontent.com/48293545/80873279-098bfc80-8cc0-11ea-9599-e86bc4e2bf6f.png">

Afterward, I created a new world with the second seed (the one shown at the Game details page), and that new world seemed to match that seed's original world.

### Issue still remains under other world generator methods

The sad part is, that the issue still remains to other world generator methods, at least it does for the "Flat" one. When testing with the "Flat" world generator, the seed did change again, but using the new changed seed for a new world, did not provide the desirable results.

So in conclusion, further tests are needed to determine if the issue still remains and it just happened to work with the world generator "Perlin", or if the real issue is with the world generators, and not the seed value alone.

**Comment by pollend** [comment]
I don't think we should be modifying the seed if that is what the seed is?

**Comment by stefaniamak** [comment]
Haven't looked into the algorithm for creating worlds from seeds, but if you apply specific modifications you want (like Perlin, or Flat), I do not find it strange for the seed to change. Rather, I find it logical if you take into consideration that the original seed was either generated completely randomly or custom wrote by the user. @pollend 

_(As I said though, I haven't looked into the actual world generator algorithm, thus these are just my theories)._

**Comment by jdrueckert** [comment]
Seed is not ignored but as @stefaniamak said it's changed along the (advanced setup) way - appended by the name of the added generator.
We definitely shouldn't change it if the user requested a specific seed.
@Cervator shall we rename the issue and reopen or create a new one?

**Comment by Cervator** [comment]
So if I'm understanding it right and remember some details correctly it isn't so much that the seed value is _changed_ it is more that there are _two_ independent seed values - and the 2nd one is the one specific to the world, simply with the world name appended to the first seed, which is meant to represent the _universe_ value - in case you have multiple worlds. Which isn't very useful right now since you can only ever spawn into one world, but _is_ correct in the long term, assuming we ever get back to finish multi-world to where you can spawn into an properly generate the other worlds :-)

So while the point _could_ be argued that we should revert to a single seed value until multi-world is complete, I'd lean toward keeping it in but making it more obvious. Going to update #3931 and #3915 accordingly, but remain open-minded to the alternative as well 👍 

With both seed values visible you should be able to replicate the world-specific seed value easily. It normally should be enough to just enter the same universe seed value since the appended world name is going to default to the same every time, unless the player specifically changes the name of the world.
